git_wrapper/log.py

```diff
@@ -82,17 +100,19 @@ def short_log_diff(self, hash_from, hash_to):
         return self.log_diff(hash_from, hash_to, "$short_hash $summary")
 
     @reference_exists('branch')
-    def grep_for_commits(self, branch, grep_for, reverse=False, path=None):
+    def grep_for_commits(self, branch, grep_for, reverse=False, path=None,
+                         log_format='%H'):
         """Returns a list of matching commits shas.
            :param str branch: which branch to grep on
            :param str grep_for: what to grep for
            :param bool reverse: whether to return in reversed order
            :param str path: path to limit the search to, optionally
-           :return: A list of commit ids
+           :param str log_format: log format output. Defaults to format:%H
+           :return: A list of resulting commit matching the pattern.
         """
         commits = []
 
-        params = [branch, "--format=format:%H", "--grep=%s" % grep_for]
+        params = [branch, "--format=%s" % log_format, "--grep=%s" % grep_for]
```

Comment:
oh, right. was due to my tests using "--pretty=format:..."
Though... in fact, I can probably drop THAT change, especially due to the way the output is later parsed. In my use-case, I'll just do 2 calls: one with the _grep in order to get commit ID(s), and one with the new show_commit() so that I get the actual information.

Reply:
note: keeping it, with a better docstring and consistency.